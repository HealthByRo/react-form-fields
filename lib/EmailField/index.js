'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _EmailInput = require('../EmailInput');

var _EmailInput2 = _interopRequireDefault(_EmailInput);

var _createFormField = require('../createFormField');

var _createFormField2 = _interopRequireDefault(_createFormField);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// $FlowFixMe
exports.default = (0, _createFormField2.default)(_EmailInput2.default);