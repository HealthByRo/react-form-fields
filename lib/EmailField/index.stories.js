'use strict';

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _storybook = require('@kadira/storybook');

var _ = require('./');

var _2 = _interopRequireDefault(_);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

(0, _storybook.storiesOf)('Fields: EmailField', module).add('input', function () {
  var props = {
    name: 'EMAIL_FIELD_NAME'
  };

  return _react2.default.createElement(_2.default, props);
}).add('input + placeholder', function () {
  var props = {
    name: 'EMAIL_FIELD_NAME',
    placeholder: 'Sample email field with placeholder'
  };

  return _react2.default.createElement(_2.default, props);
}).add('input + placeholder + error', function () {
  var props = {
    errors: ['Sample error for field'],
    name: 'EMAIL_FIELD_NAME',
    placeholder: 'Sample email field with placeholder'
  };

  return _react2.default.createElement(_2.default, props);
}).add('input + placeholder + multiple errors', function () {
  var props = {
    errors: ['First error for field', 'Second error for field'],
    name: 'EMAIL_FIELD_NAME',
    placeholder: 'Sample email field with placeholder'
  };

  return _react2.default.createElement(_2.default, props);
}).add('input + label', function () {
  var props = {
    name: 'EMAIL_FIELD_NAME',
    label: 'Sample label'
  };

  return _react2.default.createElement(_2.default, props);
}).add('input + label + error', function () {
  var props = {
    errors: ['Sample error for field'],
    name: 'EMAIL_FIELD_NAME',
    label: 'Sample label'
  };

  return _react2.default.createElement(_2.default, props);
}).add('input + label + multiple errors', function () {
  var props = {
    errors: ['First error for field', 'Second error for field'],
    name: 'EMAIL_FIELD_NAME',
    label: 'Sample label'
  };

  return _react2.default.createElement(_2.default, props);
}).add('input + label + placeholder', function () {
  var props = {
    name: 'EMAIL_FIELD_NAME',
    label: 'Sample label',
    placeholder: 'Sample email field with placeholder'
  };

  return _react2.default.createElement(_2.default, props);
}).add('input + label + placeholder + error', function () {
  var props = {
    errors: ['Sample error for field'],
    name: 'EMAIL_FIELD_NAME',
    label: 'Sample label',
    placeholder: 'Sample email field with placeholder'
  };

  return _react2.default.createElement(_2.default, props);
}).add('input + label + placeholder + mutliple errors', function () {
  var props = {
    errors: ['First error for field', 'Second error for field'],
    name: 'EMAIL_FIELD_NAME',
    label: 'Sample label',
    placeholder: 'Sample email field with placeholder'
  };

  return _react2.default.createElement(_2.default, props);
});