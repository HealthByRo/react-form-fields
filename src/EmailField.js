/* @flow */

import React, { PureComponent } from 'react';
import { LabelElement, EmailFieldElement } from './elements';
import { FieldWrapper } from './wrappers';
import Errors from './Errors';
import type { FieldProps } from './types';

export default class EmailField extends PureComponent {
  constructor(props: FieldProps) {
    super(props);

    if (!props.name || props.name.length < 1) {
      // TODO create file with errors, add MissingParamError
      throw new Error('name param is required for EmailField');
    }
  }

  composeProps(): FieldProps {
    const props = { ...this.props };

    if (!props.inputId) {
      props.inputId = this.generateInputId(props);
    }

    return props;
  }

  // TODO add some unique id at the end
  generateInputId(props: FieldProps): string {
    return `TF-${props.name}`;
  }

  props: FieldProps;

  render() {
    const props = this.composeProps();

    return (
      <FieldWrapper>
        {props.label &&
          <LabelElement {...props} />
        }
        <EmailFieldElement {...props} />
        <Errors {...this.props} />
      </FieldWrapper>
    );
  }
}
