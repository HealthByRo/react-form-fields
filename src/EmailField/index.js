/* @flow */

import EmailInput from '../EmailInput';
import createFormField from '../createFormField';

// $FlowFixMe
export default createFormField(EmailInput);
