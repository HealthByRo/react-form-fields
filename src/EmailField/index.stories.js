/* @flow */
/*

import React from 'react';
import { storiesOf } from '@kadira/storybook';
import EmailField from './';

storiesOf('Fields: EmailField', module)
  .add('input', () => {
    const props = {
      name: 'EMAIL_FIELD_NAME',
    };

    return (
      <EmailField {...props} />
    );
  })
  .add('input + placeholder', () => {
    const props = {
      name: 'EMAIL_FIELD_NAME',
      placeholder: 'Sample email field with placeholder',
    };

    return (
      <EmailField {...props} />
    );
  })
  .add('input + placeholder + error', () => {
    const props = {
      errors: ['Sample error for field'],
      name: 'EMAIL_FIELD_NAME',
      placeholder: 'Sample email field with placeholder',
    };

    return (
      <EmailField {...props} />
    );
  })
  .add('input + placeholder + multiple errors', () => {
    const props = {
      errors: [
        'First error for field',
        'Second error for field',
      ],
      name: 'EMAIL_FIELD_NAME',
      placeholder: 'Sample email field with placeholder',
    };

    return (
      <EmailField {...props} />
    );
  })
  .add('input + label', () => {
    const props = {
      name: 'EMAIL_FIELD_NAME',
      label: 'Sample label',
    };

    return (
      <EmailField {...props} />
    );
  })
  .add('input + label + error', () => {
    const props = {
      errors: ['Sample error for field'],
      name: 'EMAIL_FIELD_NAME',
      label: 'Sample label',
    };

    return (
      <EmailField {...props} />
    );
  })
  .add('input + label + multiple errors', () => {
    const props = {
      errors: [
        'First error for field',
        'Second error for field',
      ],
      name: 'EMAIL_FIELD_NAME',
      label: 'Sample label',
    };

    return (
      <EmailField {...props} />
    );
  })
  .add('input + label + placeholder', () => {
    const props = {
      name: 'EMAIL_FIELD_NAME',
      label: 'Sample label',
      placeholder: 'Sample email field with placeholder',
    };

    return (
      <EmailField {...props} />
    );
  })
  .add('input + label + placeholder + error', () => {
    const props = {
      errors: ['Sample error for field'],
      name: 'EMAIL_FIELD_NAME',
      label: 'Sample label',
      placeholder: 'Sample email field with placeholder',
    };

    return (
      <EmailField {...props} />
    );
  })
  .add('input + label + placeholder + mutliple errors', () => {
    const props = {
      errors: [
        'First error for field',
        'Second error for field',
      ],
      name: 'EMAIL_FIELD_NAME',
      label: 'Sample label',
      placeholder: 'Sample email field with placeholder',
    };

    return (
      <EmailField {...props} />
    );
  });
*/
