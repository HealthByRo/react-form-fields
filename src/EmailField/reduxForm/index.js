import EmailField from '../';
import createReduxFormField from '../../createReduxFormField';

export default createReduxFormField(EmailField);
