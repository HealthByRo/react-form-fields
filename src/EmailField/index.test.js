/* @flow */

import React from 'react';
import { shallow } from 'enzyme';
import EmailField from './';
import type { InputProps } from '../types';

describe('<EmailField />', () => {
  let wrapper;
  const defaultProps = {
    name: 'INPUT_NAME',
    id: 'INPUT_ID',
  };
  const shallowWithProps = (props: InputProps) => {
    wrapper = shallow(
      <EmailField {...props} />
    );
  };

  describe('html element', () => {
    describe('just name and inputId', () => {
      beforeEach(() => {
        shallowWithProps(defaultProps);
      });

      it('should be input', () => {
        expect(wrapper.name()).toBe('input');
      });

      it('should have [type=email]', () => {
        expect(wrapper.props().type).toEqual('email');
      });

      it('should have [name=INPUT_NAME]', () => {
        expect(wrapper.props().name).toEqual('INPUT_NAME');
      });

      it('should have default .textfield class', () => {
        expect(wrapper.hasClass('textfield')).toBe(true);
      });

      it('should not have [placeholder]', () => {
        expect(wrapper.props.placeholder).not.toBeDefined();
      });

      it('should not have [placeholder]', () => {
        expect(wrapper.props.placeholder).not.toBeDefined();
      });
    });
  });

  describe('more params', () => {
    it('should use className from props', () => {
      shallowWithProps({
        ...defaultProps,
        className: 'CSS_CLASS',
      });

      expect(wrapper.hasClass('CSS_CLASS')).toBe(true);
      expect(wrapper.hasClass('textfield')).toBe(false);
    });

    it('should use placeholder from props', () => {
      shallowWithProps({
        ...defaultProps,
        placeholder: 'PLACEHOLDER',
      });

      expect(wrapper.props().placeholder).toEqual('PLACEHOLDER');
    });
  });
});
