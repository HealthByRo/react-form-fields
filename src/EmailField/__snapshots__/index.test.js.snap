// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmailField { name } /> should render only wrapped input field with defaults 1`] = `
<div
  className="form-item"
>
  <input
    className="textfield"
    id="TF-INPUT_NAME"
    name="INPUT_NAME"
    type="email"
  />
</div>
`;

exports[`<EmailField { name, label } /> should render input and labbel 1`] = `
<div
  className="form-item"
>
  <label
    htmlFor="TF-INPUT_NAME"
  >
    INPUT_LABEL
  </label>
  <input
    className="textfield"
    id="TF-INPUT_NAME"
    name="INPUT_NAME"
    type="email"
  />
</div>
`;

exports[`<EmailField { name, label, errors: [FIRST_ERROR, SECOND_ERROR] } /> should render input and label and two errors 1`] = `
<div
  className="form-item"
>
  <label
    htmlFor="TF-INPUT_NAME"
  >
    INPUT_LABEL
  </label>
  <input
    className="textfield"
    id="TF-INPUT_NAME"
    name="INPUT_NAME"
    type="email"
  />
  <div
    className="errors-list"
  >
    <div
      className="msg msg--error"
    >
      FIRST_ERROR
    </div>
    <div
      className="msg msg--error"
    >
      SECOND_ERROR
    </div>
  </div>
</div>
`;

exports[`<EmailField { name, label, errors: [SINGLE_ERROR] } /> should render input and label and single error 1`] = `
<div
  className="form-item"
>
  <label
    htmlFor="TF-INPUT_NAME"
  >
    INPUT_LABEL
  </label>
  <input
    className="textfield"
    id="TF-INPUT_NAME"
    name="INPUT_NAME"
    type="email"
  />
  <div
    className="msg msg--error"
  >
    SINGLE_ERROR
  </div>
</div>
`;

exports[`<EmailField { name, placeholder } /> should render input with placeholder 1`] = `
<div
  className="form-item"
>
  <input
    className="textfield"
    id="TF-INPUT_NAME"
    name="INPUT_NAME"
    placeholder="INPUT_PLACEHOLDER"
    type="email"
  />
</div>
`;

exports[`<EmailField { name, placeholder, errors: [FIRST_ERROR, SECOND_ERROR] } /> should render input with placeholder and two errors 1`] = `
<div
  className="form-item"
>
  <input
    className="textfield"
    id="TF-INPUT_NAME"
    name="INPUT_NAME"
    placeholder="INPUT_PLACEHOLDER"
    type="email"
  />
  <div
    className="errors-list"
  >
    <div
      className="msg msg--error"
    >
      FIRST_ERROR
    </div>
    <div
      className="msg msg--error"
    >
      SECOND_ERROR
    </div>
  </div>
</div>
`;

exports[`<EmailField { name, placeholder, errors: [SINGLE_ERROR] } /> should render input with placeholder and single error 1`] = `
<div
  className="form-item"
>
  <input
    className="textfield"
    id="TF-INPUT_NAME"
    name="INPUT_NAME"
    placeholder="INPUT_PLACEHOLDER"
    type="email"
  />
  <div
    className="msg msg--error"
  >
    SINGLE_ERROR
  </div>
</div>
`;

exports[`<EmailField { name, placeholder, label } /> should render input with placeholder and label 1`] = `
<div
  className="form-item"
>
  <label
    htmlFor="TF-INPUT_NAME"
  >
    INPUT_LABEL
  </label>
  <input
    className="textfield"
    id="TF-INPUT_NAME"
    name="INPUT_NAME"
    placeholder="INPUT_PLACEHOLDER"
    type="email"
  />
</div>
`;

exports[`<EmailField { name, placeholder, label, errors: [FIRST_ERROR, SECOND_ERROR] } /> should render input with placeholder and label and two errors 1`] = `
<div
  className="form-item"
>
  <label
    htmlFor="TF-INPUT_NAME"
  >
    INPUT_LABEL
  </label>
  <input
    className="textfield"
    id="TF-INPUT_NAME"
    name="INPUT_NAME"
    placeholder="INPUT_PLACEHOLDER"
    type="email"
  />
  <div
    className="errors-list"
  >
    <div
      className="msg msg--error"
    >
      FIRST_ERROR
    </div>
    <div
      className="msg msg--error"
    >
      SECOND_ERROR
    </div>
  </div>
</div>
`;

exports[`<EmailField { name, placeholder, label, errors: [SINGLE_ERROR] } /> should render input with placeholder and label and single error 1`] = `
<div
  className="form-item"
>
  <label
    htmlFor="TF-INPUT_NAME"
  >
    INPUT_LABEL
  </label>
  <input
    className="textfield"
    id="TF-INPUT_NAME"
    name="INPUT_NAME"
    placeholder="INPUT_PLACEHOLDER"
    type="email"
  />
  <div
    className="msg msg--error"
  >
    SINGLE_ERROR
  </div>
</div>
`;
